### PR TITLE
feat: upgrade ApisixRoute v2beta2 apiversion.

### LINKED ISSUE #692: Breaking changes: deprecated the `spec.backend` field in ApisixRoute
### Issue description

In the current ApisixRoute CRD definition, there are both `spec.backend` and `spec.backends`

https://github.com/apache/apisix-ingress-controller/blob/957c31522e1b1e5f8ef9cab7eb244473a4e0f675/samples/deploy/crd/v1beta1/ApisixRoute.yaml#L173-L217

And `spec.backends` can completely cover the usage scenarios of `spec.backend`.  Currently maintaining these two fields at the same time has caused a certain burden on us.
For example, when the resource cannot be obtained directly through kubectl, the corresponding information can be displayed. For details, please see https://github.com/apache/apisix-ingress-controller/pull/626#discussion_r686714645

**I suggest that we upgrade ApisixRoute to v2beta2 version as soon as possible, and deprecate `spec.backend` in this version.**

### PR DESCRIPTION
**Pull request by tao12345666333**
**Notice:**

* remove `spec.backend`
* remove `spec.tcp`

Signed-off-by: Jintao Zhang <zhangjintao9020@gmail.com>

Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues

Closes #692 
___
### Bugfix
- Description

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

___
### Backport patches
- Why need to backport?

- Source branch

- Related commits and pull requests

- Target branch

### CODE REVIEW

**Comment by tao12345666333** [comment]
need to be done:

* add new test cases cover this version.
* update the document and no longer use related fields

**Comment by tokers** [comment]
I think we should try to generate crd from their data structure definition in the future, to avoid writing them by hands.

**Review by tokers — APPROVED** [review]

**Comment by tao12345666333** [comment]
Yes, we have created an issue for tracking. #515   We can deal with it in the next version.

By the way, I suggest to unify the existing CRD apiversion in this project, which can reduce the learning cost of users and facilitate unified management.

**Review by gxthrj — APPROVED** [review]

**Comment by gxthrj** [comment]
@tao12345666333  there are still some lint errors.

**Comment by tao12345666333** [comment]
yep. just edit on GitHub codespace. I will fix it ASAP

**Comment by gxthrj** [comment]
Please check lint error.

**Review by lingsamuel — APPROVED** [review]
LGTM, just fix linter errors

**Comment by tao12345666333** [comment]
```
475183_apisix-service-e2e-test_9180", "error": "not found"}                                                                                                                           
panic: runtime error: invalid memory address or nil pointer dereference                                                                                                               
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x15fdb81]                                                                                                               
                                                                                                                                                                                      
goroutine 360 [running]:                                                                                                                                                              
github.com/apache/apisix-ingress-controller/pkg/kube/translation.(*translator).getServiceClusterIPAndPort(0xc0005e0080, 0x0, 0xc0009df7d0, 0x2a, 0xc00085cca8, 0x0, 0x0, 0x0, 0x0)    
        /build/pkg/kube/translation/util.go:38 +0x81                                                                                                                                  
github.com/apache/apisix-ingress-controller/pkg/kube/translation.(*translator).translateHTTPRoute(0xc0005e0080, 0xc0002b8e00, 0xc00015c128, 0x1cb4bee, 0x1cb4bee)                             /build/pkg/kube/translation/apisix_route.go:439 +0x860                                                                                                                        
github.com/apache/apisix-ingress-controller/pkg/kube/translation.(*translator).TranslateRouteV2alpha1(0xc0005e0080, 0xc00015c128, 0x1a, 0xc000a35c01, 0xd)
        /build/pkg/kube/translation/apisix_route.go:103 +0x7c                                                                                                                         github.com/apache/apisix-ingress-controller/pkg/ingress.(*apisixRouteController).sync(0xc0003e9980, 0x1fd6000, 0xc0005d2300, 0xc0006ea960, 0x1d69390, 0x0)
        /build/pkg/ingress/apisix_route.go:149 +0x1d9a                                                                                                                                
github.com/apache/apisix-ingress-controller/pkg/ingress.(*apisixRouteController).runWorker(0xc0003e9980, 0x1fd6000, 0xc0005d2300)                                                             /build/pkg/ingress/apisix_route.go:79 +0x57     
created by github.com/apache/apisix-ingress-controller/pkg/ingress.(*apisixRouteController).run              
        /build/pkg/ingress/apisix_route.go:68 +0x245         
```

when using `v2alpha1`  ApisixRoute to run e2e, I got a panic.

File: test/e2e/ingress/tcp.go [hunk 1]
```diff
@@ -1,71 +0,0 @@
...
-	"fmt"
-	"time"
-
-	"github.com/onsi/ginkgo"
-	"github.com/stretchr/testify/assert"
-
-	"github.com/apache/apisix-ingress-controller/test/e2e/scaffold"
-)
-
-var _ = ginkgo.Describe("ApisixRoute Testing", func() {
```
Comment (gxthrj) [inline test/e2e/ingress/tcp.go:27]: Why delete this ？

Comment (tao12345666333) [reply test/e2e/ingress/tcp.go:27]: We have completely deleted `spec.tcp` and use `spec.stream` instead

**Review by tokers — APPROVED** [review]

**Pull request merged**